pyspark-iris/{{ cookiecutter.repo_name }}/conf/base/catalog.yml

```diff
@@ -5,22 +5,99 @@
 #
 # We support interacting with a variety of data stores including local file systems, cloud, network and HDFS
 #
+# An example data set definition can look as follows:
+#
+#bikes:
+#  type: pandas.CSVDataSet
+#  filepath: "data/01_raw/bikes.csv"
+#
+#weather:
+#  type: spark.SparkDataSet
+#  filepath: s3a://your_bucket/data/01_raw/weather*
+#  file_format: csv
+#  credentials: dev_s3
+#  load_args:
+#    header: True
+#    inferSchema: True
+#  save_args:
+#    sep: '|'
+#    header: True
+#
+#scooters:
+#  type: pandas.SQLTableDataSet
+#  credentials: scooters_credentials
+#  table_name: scooters
+#  load_args:
+#    index_col: ['name']
+#    columns: ['name', 'gear']
+#  save_args:
+#    if_exists: 'replace'
+#    # if_exists: 'fail'
+#    # if_exists: 'append'
+#
 # The Data Catalog supports being able to reference the same file using two different DataSet implementations
 # (transcoding), templating and a way to reuse arguments that are frequently repeated. See more here:
 # https://kedro.readthedocs.io/en/stable/data/data_catalog.html
 #
 # This is a data set used by the iris classification example pipeline provided with this starter
 # template. Please feel free to remove it once you remove the example pipeline.
+
 example_iris_data:
   type: spark.SparkDataSet
   filepath: data/01_raw/iris.csv
   file_format: csv
   load_args:
-    header: true
-    inferSchema: true
+    header: True
+    inferSchema: True
   save_args:
-    sep: ","
-    header: true
+    sep: '|'
+    header: True
+
+# We need to set mode to 'overwrite' in save_args so when saving the dataset it is replaced each time it is run
+# for all SparkDataSets.
+X_train@pyspark:
+  type: spark.SparkDataSet
+  filepath: data/02_intermediate/X_train.parquet
+  file_format: parquet
```

Comment:
I think we don't need to specify `file_format: parquet` since it's the default. I'd just leave it out of all the catalog definitions.